azurerm/resource_arm_application_insights_analytics_item.go

```diff
@@ -0,0 +1,219 @@
+				Optional: true,
+			},
+
+			"time_created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"time_modified": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmApplicationInsightsAnalyticsItemCreate(d *schema.ResourceData, meta interface{}) error {
+	return resourceArmApplicationInsightsAnalyticsItemCreateUpdate(d, meta, false)
+}
+func resourceArmApplicationInsightsAnalyticsItemUpdate(d *schema.ResourceData, meta interface{}) error {
+	return resourceArmApplicationInsightsAnalyticsItemCreateUpdate(d, meta, true)
+}
+func resourceArmApplicationInsightsAnalyticsItemCreateUpdate(d *schema.ResourceData, meta interface{}, overwrite bool) error {
+	client := meta.(*ArmClient).appInsights.AnalyticsItemsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	appInsightsID := d.Get("application_insights_id").(string)
+
+	id, err := azure.ParseAzureResourceID(appInsightsID)
+	if err != nil {
+		return fmt.Errorf("Error parsing resource ID: %s", err)
+	}
+
+	appInsightsName := id.Path["components"]
+
+	itemID := d.Id()
+	name := d.Get("name").(string)
+	content := d.Get("content").(string)
+	scopeName := d.Get("scope").(string)
+	typeName := d.Get("type").(string)
+	functionAlias := d.Get("function_alias").(string)
+
+	itemType := insights.ItemType(typeName)
+	itemScope := insights.ItemScope(scopeName)
+	properties := insights.ApplicationInsightsComponentAnalyticsItem{
+		ID:      &itemID,
+		Name:    &name,
+		Type:    itemType,
+		Scope:   itemScope,
+		Content: &content,
+	}
+	if functionAlias != "" {
+		properties.Properties = &insights.ApplicationInsightsComponentAnalyticsItemProperties{
+			FunctionAlias: &functionAlias,
+		}
+	}
+
+	var itemScopePath insights.ItemScopePath
+	if itemScope == insights.ItemScopeUser {
+		itemScopePath = insights.MyanalyticsItems
+	} else {
+		itemScopePath = insights.AnalyticsItems
+	}
+	result, err := client.Put(ctx, resourceGroupName, appInsightsName, itemScopePath, properties, &overwrite)
+	if err != nil {
+		return fmt.Errorf("Error Putting Application Insights Analytics Item %s (Resource Group %s, App Insights Name: %s): %s", name, resourceGroupName, appInsightsName, err)
+	}
+
+	d.SetId(*result.ID)
+
+	return resourceArmApplicationInsightsAnalyticsItemRead(d, meta)
+}
+
+func resourceArmApplicationInsightsAnalyticsItemRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).appInsights.AnalyticsItemsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	appInsightsID := d.Get("application_insights_id").(string)
+	scopeName := d.Get("scope").(string)
+	itemID := d.Id()
+
+	id, err := azure.ParseAzureResourceID(appInsightsID)
+	if err != nil {
+		return fmt.Errorf("Error parsing resource ID: %s", err)
+	}
+
+	appInsightsName := id.Path["components"]
+	name := d.Get("name").(string)
```

Comment:
(same here) can we parse this out of the ID, since this won't be available during an import?

Reply:
I've updated this to use a generated ID with the required properties embedded